pyk/src/pyk/cterm/cterm.py

```diff
@@ -93,10 +93,12 @@ def _check_config(config: KInner) -> None:
             raise ValueError(f'Expected cell label, found: {config}')
 
     @staticmethod
-    def _normalize_constraints(constraints: Iterable[KInner]) -> tuple[KInner, ...]:
+    def _normalize_constraints(constraints: Iterable[KInner], maintain_order: bool = False) -> tuple[KInner, ...]:
```

Comment:
Consider extracting the flattening logic into a helper function. Then `maintain_order` can be removed, and `CSubst` won't have to call the private method of another class.